### ISSUE TITLE
About the questions in the paper

### ISSUE THREAD

**Issue by yzrs** [kingsj0405/DIFE#1]
Hi, in Figure3 of the experimental results of keypoints transferring in the paper，is the showing Source Image the original input image ？or the synthesized image output by StyleGAN2?
By the way, to be sure, the DIFE encoder is trained both by distance to CSE and by semantic matching to pseudo-paired images, right?

**Comment by kingsj0405** [comment]
Hello @yzrs. Thank you for your interest about our work :)

> Hi, in Figure3 of the experimental results of keypoints transferring in the paper，is the showing Source Image the original input image ？or the synthesized image output by StyleGAN2?

On the Figure 3 on main paper and Figure 4 on [supplementary material](https://openreview.net/attachment?id=m67FNFdgLO9&name=supplementary_material), images comes from human face keypoint dataset (MAFL, WFLW) and animal face keypoint dataset (AP-10K, AnimalWeb). Because we need annotated facial keypoint to transfer, we can't utilize the synthesized data from StyleGAN2.

> By the way, to be sure, the DIFE encoder is trained both by distance to CSE and by semantic matching to pseudo-paired images, right?

Yes. DIFE encoder is pre-trained with CSE feature and StyleGAN2 feature at first. Then trained with synthedsized pesudo paired data by semantic matching loss. For your information I upload the `train_unrefactored.py` to current repository. The related code is [here](https://github.com/kingsj0405/DIFE.private/blob/0ab0bc57b79161a54d0725deee030fedbaac396e/scripts/train_ours_interspecies.py#L146).

**Comment by yzrs** [comment]
Ah I see,thanks a lot
